schema_based_on_diagram.sql

```diff
@@ -0,0 +1,54 @@
+CREATE DATABASE hospital_db;
+
+CREATE TABLE patients(
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(150),
+  date_of_birth DATE
+);
+
+CREATE TABLE medical_histories(
+  id SERIAL PRIMARY KEY, 
+  admitted_at TIMESTAMP, 
+  patient_id INTEGER, 
+  FOREIGN KEY (patient_id) REFERENCES patients(id), 
+  status VARCHAR(255)
+);
+
+CREATE TABLE invoices(
+  id SERIAL PRIMARY KEY, 
+  total_amount DECIMAL, 
+  generated_ad TIMESTAMP, 
+  payed_at TIMESTAMP, 
+  medical_history_id INTEGER, 
+  FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id)
+);
+
+CREATE TABLE teatments(
+  id SERIAL PRIMARY KEY,
+  type VARCHAR(255),
+  name VARCHAR(255)
+);
+
+CREATE TABLE invoice_items(
+  id SERIAL PRIMARY KEY,
+  unit_price DECIMAL,
+  quantity INTEGER,
+  total_price DECIMAL,
+  invoice_id INTEGER,
+  treatment_id INTEGER,
+  FOREIGN KEY (invoice_id) REFERENCES invoices(id),
+  FOREIGN KEY (treatment_id) REFERENCES treatments(id)
+);
+
+CREATE TABLE medical_histories_has_treatment(
+  medical_history_id INTEGER, 
+  treatment_id INTEGER, 
+  FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id), 
+  FOREIGN KEY (treatment_id) REFERENCES treatments(id)
```

Comment:
Great progress so far! You've done a lot of work. However, there are a couple of adjustments needed:

- [ ] First, you should create indexes for all your foreign keys as recommended by the previous reviewer. Currently, there are no indexes created for these two foreign keys. Please consider updating your code to meet the project requirements by adding indexes for all your foreign keys.

- [ ] Second, there seems to be a typo error. Your table name is `teatments`, but you referred to it as `treatments` in your code. You need to either change your table name to match your references or update the references to ensure there's no conflict. 

Making these changes will help ensure your database aligns with the project requirements and that there are no errors due to typos. Keep up the good work! 👍🗃️